websites/W/Warriorcats/presence.ts

```diff
@@ -0,0 +1,270 @@
+			presence.setActivity(presenceData);
+
+			break;
+		}
+		case "/erin-hunter/": {
+			presenceData.details = "Viewing";
+			presenceData.buttons = [
+				{ label: "Current Page", url: document.location.href },
+				{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+			];
+			presenceData.state = "stuff about Erin Hunter";
+			presenceData.largeImageKey =
+				"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+			presence.setActivity(presenceData);
+
+			break;
+		}
+		case "/welt-der-warrior-cats/": {
+			presenceData.details = "Viewing";
+			presenceData.buttons = [
+				{ label: "Current Page", url: document.location.href },
+				{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+			];
+			presenceData.state = "the Warrior Cats World";
+			presenceData.largeImageKey =
+				"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+			presence.setActivity(presenceData);
+
+			break;
+		}
+		case "/zeitstrahl/": {
+			presenceData.details = "Viewing";
+			presenceData.state = "the Warrior Cats LORE";
+			presenceData.buttons = [
+				{ label: "Current Page", url: document.location.href },
+				{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+			];
+			presenceData.largeImageKey =
+				"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+
+			break;
+		}
+		case "/wiki/": {
+			presenceData.details = "Viewing";
+			presenceData.state = "the Warrior Cats Wiki";
+			presenceData.buttons = [
+				{ label: "Current Page", url: document.location.href },
+				{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+			];
+			presenceData.largeImageKey =
+				"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+
+			break;
+		}
+		default: {
+			presenceData.details = "Viewing an unsupported page";
+			presenceData.buttons = [
+				{ label: "Current Page", url: document.location.href },
+				{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+			];
+			presenceData.largeImageKey =
+				"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+		}
+	}
+
+	if (document.location.pathname.startsWith("/staffel-")) {
+		presenceData.details = `Viewing Staffel ${
+			document.location.pathname.split("/")[1].split("-")[1]
+		}`;
+
+		presenceData.largeImageKey =
+			"https://www.warriorcats.de/wp-content/uploads/9783407759085-web.jpg";
+		presenceData.buttons = [
+			{ label: "Current Page", url: document.location.href },
+			{ label: "Warrior Cats Germany", url: "https://www.warriorcats.de/" },
+		];
+	}
+
+	// https://www.warriorcats.de/beute/staffel-viii-band-2-wolken/
+	// extract Staffel number and book number and book name from URL
+	// like this Staffel 8 Band 2 Wolken
+	// and then set the presence state to Currently Staffel 8 Band 2 Wolken
+
+	async function getStaffelName(postNumber: string) {
```

Comment:
These functions do not do anything `async`. Also, for better performance/readability, these functions should be defined outside of the `UpdateData` loop.